drivers/include/periph/dac.h

```diff
@@ -102,6 +103,30 @@ enum {
  */
 int8_t dac_init(dac_t line);
 
+/**
+ * @brief   The callback that will be called when the end of the current sample buffer
+ *          has been reached.
+ *          Should be used to start filling the next sample buffer.
+ *
+ * @note    Will be called in interrupt context. Only use the callback to signal a
+ *          thread. Don't directly fill the sample buffer in the callback.
+ */
+typedef void (*dac_cb_t)(void *arg);
```

Comment:
I also noticed an inconsistency, since we should *always* call the callback when the second buffer can be queue. That means also after the first call to `dac_play()` - otherwise we are not using the double buffering at all.
(Still have to test that latest change)